tests/test_base.py

```diff
@@ -787,6 +787,46 @@ def test_current_tasks_eager(self):
 class ReliabilityTestCase(BaseTestCase):
     """ Test behavior if things go wrong. """
 
+    def _test_expired_task(self, task, expected_state):
+        """
+        Ensure the given task ends up in the expected state if the worker is
+        killed prematurely.
```

Comment:
Could add a note that `task` has to run for longer than `DELAY` for this test to work.